manticore/manticore.py

```diff
@@ -203,9 +214,9 @@ def __init__(self, path_or_state, argv=None, workspace_url=None, policy='random'
             self._initial_state = make_initial_state(path_or_state, argv=argv, **kwargs)
         elif isinstance(path_or_state, State):
             self._initial_state = path_or_state
-        else:
-            raise TypeError("Manticore must be intialized with either a State or a path to a binary")
 
+        if not isinstance(self._initial_state, State):
```

Comment:
Move to top of func and update to: `if not isinstance(path_or_state, (State, str))`? self._initial_state isn't technically an argument

Reply:
But I wanted to be sure the object created by make_initial_state is a State also.

Reply:
under what situations can `make_initial_state` return a non-`State` object?

Reply:
Hopefully none. That would be a bug. Are we sure there is no bug in current or future implementations of make_initial_state and its constituents parts ?

This one check guards for both a parameter or a make_initial_state bug.